test/extensions/clusters/eds/eds_test.cc

```diff
@@ -1403,6 +1403,133 @@ TEST_F(EdsTest, EndpointMovedToNewPriority) {
     EXPECT_FALSE(hosts[1]->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC));
     EXPECT_FALSE(hosts[1]->healthFlagGet(Host::HealthFlag::PENDING_DYNAMIC_REMOVAL));
   }
+
+  // 1 -> 2
```

Comment:
I think this bug should have its specific test (and please add a comment for the regression issue at the top of the test).